pallets/liquidity-pools-gateway/src/tests.rs

```diff
@@ -42,812 +48,4185 @@ mod utils {
+						let session_id = 1;
+						let domain_address = DomainAddress::EVM(1, [1; 20]);
+						let router_id = ROUTER_ID_1;
+						let gateway_message = GatewayMessage::Inbound {
+							domain_address: domain_address.clone(),
+							message: message.clone(),
+							router_id: router_id.clone(),
+						};
+
+						Routers::<Runtime>::set(
+							BoundedVec::<_, _>::try_from(vec![router_id.clone()]).unwrap(),
+						);
+						SessionIdStore::<Runtime>::set(session_id);
+						PendingInboundEntries::<Runtime>::insert(
+							message_proof,
+							router_id,
+							InboundEntry::Proof(ProofEntry {
+								session_id,
+								current_count: 0,
+							}),
+						);
+
+						let (res, _) = LiquidityPoolsGateway::process(gateway_message);
+						assert_noop!(res, Error::<Runtime>::ExpectedMessageProofType);
+					});
+				}
+			}
+
+			mod two_routers {
+				use super::*;
+
+				mod success {
+					use super::*;
+
+					lazy_static! {
+						static ref TEST_DATA: Vec<RouterMessage> = vec![
+							(ROUTER_ID_1, Message::Simple),
+							(ROUTER_ID_2, Message::Proof(MESSAGE_PROOF)),
+						];
+					}
+
+					mod two_messages {
+						use super::*;
+
+						const MESSAGE_COUNT: usize = 2;
+
+						#[test]
+						fn success() {
+							let expected_results: HashMap<Vec<RouterMessage>, ExpectedTestResult> =
+								HashMap::from([
+									(
+										vec![
+											(ROUTER_ID_1, Message::Simple),
+											(ROUTER_ID_1, Message::Simple),
+										],
+										ExpectedTestResult {
+											message_submitted_times: 0,
+											expected_storage_entries: vec![
+												(
+													ROUTER_ID_1,
+													Some(
+														MessageEntry {
+															session_id: TEST_SESSION_ID,
+															domain_address: TEST_DOMAIN_ADDRESS,
+															message: Message::Simple,
+															expected_proof_count: 2,
+														}
+														.into(),
+													),
+												),
+												(ROUTER_ID_2, None),
+											],
+										},
+									),
+									(
+										vec![
+											(ROUTER_ID_2, Message::Proof(MESSAGE_PROOF)),
+											(ROUTER_ID_2, Message::Proof(MESSAGE_PROOF)),
+										],
+										ExpectedTestResult {
+											message_submitted_times: 0,
+											expected_storage_entries: vec![
+												(ROUTER_ID_1, None),
+												(
+													ROUTER_ID_2,
+													Some(
+														ProofEntry {
+															session_id: TEST_SESSION_ID,
+															current_count: 2,
+														}
+														.into(),
+													),
+												),
+											],
+										},
+									),
+									(
+										vec![
+											(ROUTER_ID_1, Message::Simple),
+											(ROUTER_ID_2, Message::Proof(MESSAGE_PROOF)),
+										],
+										ExpectedTestResult {
+											message_submitted_times: 1,
+											expected_storage_entries: vec![
+												(ROUTER_ID_1, None),
+												(ROUTER_ID_2, None),
```

Comment:
Instead of `None`, should it then be removed? Could then the entry be there forever?

Reply:
We don't store options in storage, the value you're referring to gets compared to the output of `get` on the storage here:
https://github.com/centrifuge/centrifuge-chain/blob/584e7e8eb4333942a1093bc5f3bb2c9b06a5114c/pallets/liquidity-pools-gateway/src/tests.rs#L1038

Reply:
So that should mean that the storage is empty.

Reply:
Ah great! Thanks!